### ISSUE TITLE
currency and priceAsDecimal field missing

### ISSUE THREAD

**Issue by dpthinkwik** [AlexDisler/cordova-plugin-inapppurchase#191]
```
inAppPurchase
  .getProducts(['monthly_subscription'])
  .then(function (products) {
    console.log(products[0]);
  })
  .catch(function (err) {
    console.log(err);
  });
```

above is the code i have tried but i am seeing there are currency and priceAsDecimal field missing from response.

![screen shot 2017-11-28 at 12 41 27 pm](https://user-images.githubusercontent.com/18225876/33306669-82aa4e24-d439-11e7-83ac-8effabe9f037.png)

And this happening not only for subscription based plan but also with non-consumables as well.

So is there any config issue from my side or plugin has something to do with it. 

So please help me with it.

**Comment by dpthinkwik** [comment]
i finally made it work by changing PaymentsPlugin.m file with below code.

```
NSLocale* storeLocale = product.priceLocale;
NSString *storeCurrency = (NSString*)CFLocaleGetValue((CFLocaleRef)storeLocale, kCFLocaleCurrencyCode);
```
right after loop for products. And also added currency field in response. So complete code will look as below screenshot.

![screen shot 2017-11-28 at 2 44 34 pm](https://user-images.githubusercontent.com/33647326/33311439-d6c624e0-d44a-11e7-9354-d7b6b326f43f.png)

**Comment by dpthinkwik** [comment]
Still i will leave this issue open as i think this is still not proper solution. So want to hear from others as well.

**Comment by moritzgloeckl** [comment]
Hmm it's strange that it's not working, because we're iterating over `SKProduct`, and https://developer.apple.com/documentation/storekit/skproduct has a price as a decimal number...

**Comment by dpthinkwik** [comment]
@moritzgloeckl : Yes actually that is returning but issue is plugin is not getting those necessary details and also javascript part is also missing those code.
